### ISSUE TITLE
[3.2.3] Texture filtering not going off

### ISSUE THREAD

**Issue by arisu0** [godotengine/godot#45206]
**Godot version:**
Godot 3.2.3

**OS/device including version:**
Gentoo Linux, Nvidia gtx 850m (nvidia-drivers) gles3
USE flags:
[image](https://user-images.githubusercontent.com/39315312/104717589-d1332e00-5729-11eb-9a92-3a13515c6813.png)

**Issue description:**
Nearest neighbor not applying when filter is disabled on import, textures are always filtered.

**Minimal reproduction project:**
[game.zip](https://github.com/godotengine/godot/files/5820085/game.zip)
Behavior in my system:
[image](https://user-images.githubusercontent.com/39315312/104717798-28d19980-572a-11eb-80cb-166f6efab5df.png)

**Comment by Calinou** [comment]
This is because the wooden table is using an *embedded* texture in `table.glb` rather than `wood.png` on the filesystem. To solve this, you have to make the table model use the external texture rather than the internal one.

**Comment by arisu0** [comment]
Yeah, you were right. Sorry about that and thank you!

**Comment by TheTophatDemon** [comment]
The GLTF specs say that there are sampler objects that define texture filtering in the files, and the model file that @arisu0 provided contains magFilter of 9728 (NEAREST) and minFilter of 9984 (NEAREST_MIPMAP_NEAREST). This issue can be worked around, but shouldn't the imported model be getting texture filter information from the file anyways?

**Comment by Calinou** [comment]
Note that this needs different PRs to be fixed in `master` and `3.x`, since the glTF importer code differs and texture flags are set in the material on `master` (instead of the Texture resource).

**Comment by elvisish** [comment]
It also doesn't import the mipmap settings in the default, they always default to on.

**Comment by TheTophatDemon** [comment]
I've created a fork to try and fix this for the 3.x branch. This is my first time, and I'm pretty busy, so this may take awhile. Feel free to work ahead of me if necessary.
I've noticed that the GLTF module is not parsing the "samplers" array (the one under the root for textures, not the ones for the animations). This array includes information for texture filtering and wrapping, and so a solution would need to add code to parse and store that.
Also, the only way to configure mipmaps, as far as I can tell in the GLTF specs, is to choose the correct texture filter (NEAREST for no mipmaps, NEAREST_MIPMAP_[LINEAR/NEAREST] for mipmaps). The mipmaps are not stored in the file but are supposed to be generated by the program.

**Comment by fire** [comment]
@TheTophatDemon Please submit for review.

**Comment by MrJoshBowman** [comment]
Yep, the current lack of being able to set non-filtered textures visually breaks my tile-based models significantly on export.

**Comment by MrJoshBowman** [comment]
So i added this into the .gltf file and it fixed everything for me.

```
"textures": [{ "sampler": 0, "source": 0 }],
"samplers": [
  {
    "magFilter": 9728,
    "minFilter": 9728,
    "wrapS": 10497,
    "wrapT": 10497
  }
]
```
Also wondering why .gltf has separated image and mesh files? It's far easier to handle the single .gltf file with the embedded texture.

**Comment by elvisish** [comment]
Is this being fixed for 3.6? It’s nice to import a glb into Godot 4.0 and have it respect the filter settings (or easily override them in the project settings) but I still have to use the clunky gltf with separate material imports in 3.5.

**Comment by Calinou** [comment]
> Is this being fixed for 3.6?

There's a draft pull request, but it appears to be abandoned by its author: https://github.com/godotengine/godot/pull/54044

**Comment by elvisish** [comment]
> > Is this being fixed for 3.6?
> 
> 
> 
> ~~There's a draft pull request, but it appears to be abandoned by its author: https://github.com/godotengine/godot/pull/54044~~
> 
> 
> 
> **Edit:** This was implemented in https://github.com/godotengine/godot/pull/66856, which was merged after 3.5 was released. Can you try [building the latest `3.x` branch from source](https://docs.godotengine.org/en/stable/development/compiling/index.html)?

I've never been able to build from source, it doesn't look as though it's been added to the latest 3.x release though?

**Comment by lyuma** [comment]
Can someone using 3.x test if this is still an issue?

**Comment by TheTophatDemon** [comment]
@lyuma 
Just tried it on 3.x branch and release 3.5.3. The bug is still present.
